### PR TITLE
change mgmt. example generation to use `ClientFactory`

### PR DESCRIPTION
**Pull request by tadelesh**
Refer [this](https://github.com/Azure/autorest.go/blob/9fd8355cc8ec62ba33c8cd62fc2aa40894650483/packages/autorest.gotest/test/integrationtest/output/compute/armcompute/virtualmachines_client_example_test.go#L25-L34) for new client creation and usage pattern.

### CODE REVIEW

**Review by ArcturusZhang — APPROVED** [review]

File: packages/autorest.gotest/src/generator/exampleGenerator.ts [hunk 1]
```diff
@@ -8,10 +8,36 @@ import { Config } from '../common/constant';
 import { ExampleModel, MockTestDefinitionModel } from '@autorest/testmodeler/dist/src/core/model';
 import { MockTestDataRender } from './mockTestGenerator';
 import { ParameterOutput } from '../common/model';
-import { camelCase } from 'lodash';
+import { camelCase, values } from 'lodash';
 import _ = require('lodash');
+import { ObjectSchema, Parameter, SchemaType } from '@autorest/codemodel';
+import { sortParametersByRequired } from '@autorest/go/dist/generator/helpers';
```
Comment (lirenhe) [inline packages/autorest.gotest/src/generator/exampleGenerator.ts:14]: Could you use prettier to format the code?

Comment (lirenhe) [reply packages/autorest.gotest/src/generator/exampleGenerator.ts:14]: I am not sure about the current style

File: packages/autorest.gotest/test/integrationtest/output/agrifood/armagrifood/extensions_client_example_test.go [hunk 2]
```diff
@@ -41,11 +41,11 @@ func ExampleExtensionsClient_Get() {
 		log.Fatalf("failed to obtain a credential: %v", err)
 	}
 	ctx := context.Background()
-	client, err := armagrifood.NewExtensionsClient("11111111-2222-3333-4444-555555555555", cred, nil)
+	clientFactory, err := armagrifood.NewClientFactory("<subscription-id>", cred, nil)
```
Comment (lirenhe) [inline packages/autorest.gotest/test/integrationtest/output/agrifood/armagrifood/extensions_client_example_test.go:44]: Do we need to create clientFactory everywhere?

File: packages/autorest.gotest/src/generator/exampleGenerator.ts (hunk 1 above)
Comment (tadelesh) [reply packages/autorest.gotest/src/generator/exampleGenerator.ts:14]: Actually, the code has already been formatted. When migrating to this repo, I follow the exists lint rule and it seems quite loosen.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**